waddrmgr/address.go

```diff
@@ -366,6 +407,7 @@ type scriptAddress struct {
 	scriptCT        []byte
 	scriptMutex     sync.Mutex
 	used            bool
+	watchingOnly    bool
```

Comment:
Same deal here as with the p2pkh address.  We could define a watching-only p2sh address as one with a nil scriptEncrypted field.